src/DataSets/DataSets.component.js

```diff
@@ -344,6 +397,18 @@ const DataSets = React.createClass({
                                 onClose={listActions.hideDetailsBox}
                             />
                         : null}
+                    {
+                         this.state.logsObject ? (
+                             <Dialog
```

Comment:
I don't want to slow down the release, but the amount of code involving logs probably deserve a separate component. You'd write here `<LogsDialog ...` and some methods and logic would end up there (getLogs, openLogs, ...). This would also avoid the `this.setState({logsObject: true})`, which is a bit of hack.